Changes to page classes

Apogate opened the pull request:
- Removes everything that was hardcoded
- Removes admin head function, everything goes through the single
  generate() function.

cyggy committed:
Changes to page classes

- Removes everything that was hardcoded
- Removes admin head function, everything goes through the single
generate() function.

RePod commented:
Please include anything that also implemented these changes, but save Admin for its PR.

Apogate commented:
All the changes were in catalog, archive, and /admin/* classes

RePod commented:
````PHP
$dat .= "<script src='" . JS_PATH . "/{$script}' type='text/javascript'></script>";
````

{ } in a double-quoted (") line forces evaluation of the variable if it may conflict. Are { } required?

RePod commented on _core/page/head.php:
```diff
@@ -43,111 +41,77 @@ function generate() {
-            }
-            $dat .= "[<a href='" . PHP_ASELF . "?mode=logout'>" . S_LOGOUT . "</a>]</div>";
+        foreach($this->info['js']['script'] as $script) //Adding scripts to the head
+            $dat .= "<script src='" . JS_PATH . "/{$script}' type='text/javascript'></script>";
```
``` PHP
$dat .= "<script src='" . JS_PATH . "/{$script}' type='text/javascript'></script>";
```

{ } in a double-quoted (") line forces evaluation of the variable if it may conflict. Are { } required? 

For instance, `{$script}omatic` would be required to evaluate `$script` and not `$scriptomatic`.

Apogate replied:
this is intentional, just thought it looks cleaner/readable than breaking the string up like:

`$dat .= "<script src='" . JS_PATH . "/" . $script . "' type='text/javascript'></script>";`

You're going to see a lot of that in future prs, it's a new habit i picked up

RePod replied:
But if you're using "" anyway, there's no reason to break it out to `"/" . $script . "`. Maybe if using single quotes, but that's it.

No real argument on the readability, just a bit extra to accomplish the same thing.

RePod commented:
General tip: Notepad++ has a default macro on `Alt+Shift+S` that removes all trailing whitespace (tabs, spaces) then saves the document.

Trailing whitespace can build up easily when adding or re/moving new lines and just add a bit to the filesize and diff (a single new space at the end marks an entire line as changed).

RePod commented:
>All the changes were in catalog, archive, and /admin/* classes

Does this break actual page generation? I'm pretty sure Log uses this class too.

Apogate commented:
Without the changes in the next PR i'm about to open, yes, it breaks page generation.

RePod commented:
Seems like they'd be included here then.

Apogate commented:
wew

cyggy committed:
Fileboard, Log changes, API and Catalog changes.

- 4chan/8ch compliant API. We currently support *every* value available
in 4chan's api doc sans the capcode field.

- Native archiving framework. Isn't hooked into regist yet since
regist's fate is still being DEBATED.

- Fileboard support, work in progress
- Currently doesn't bump, or accept filetypes outside webm, gif, png,
jpeg. We're getting there.

- Catalog was rewritten to allow toggling between static, server side
generation, or client side js generation using the API. Catalog image
class HTML was tweaked to make it neater.
- Weird CSS bug with static generation right now, still working on that.

- Log's changes include using the page classes, integration with file
boards, and API implementation. API could probably be polished up a bit
down the road.

- RSS feeds now available for the 3 people that use them.

RePod commented:
No particular comment on the latest commit.

*Grumble grumble bad API grumble grumble.*

Apogate commented on _core/log/log.php:
```diff
@@ -346,98 +437,23 @@ function update_cache($revalidate = false) {
         $this->cache = $log;
     }
 
-    function generate($type, $no, $inIndex = false) {
```
don't freak out. they were all moved into /index/index.php

Apogate commented:
out of date with my personal branch.
